src/naf-dialog-adapter.js

```diff
@@ -192,161 +188,122 @@ export default class DialogAdapter {
     }
   }
 
-  /**
-   * This applies to the ICE restart method below.
-   * Why do we need new ICE servers? In case the ICE fails we need to re-negotiate new candidates
-   * but most likely our TURN credentials have expired so we need to request new credentials and
-   * update the TURN servers to be able to gather new candidates.
-   * Firefox doesn't support hot updating ICE servers through mediasoup's updateIceServers (which
-   * internally uses setConfiguration):
-   * https://developer.mozilla.org/en-US/docs/Web/API/RTCPeerConnection/setConfiguration
-   * So we recreate the transports/consumer/producers through reconnect to force setting
-   * new ICE servers for both send and receive transports.
-   */
+  async iceRestart(transport) {
+    // Force an ICE restart to gather new candidates and trigger a reconnection
+    this.emitRTCEvent(
+      "log",
+      "RTC",
+      () => `Restarting ${transport.id === this._sendTransport.id ? "send" : "receive"} transport ICE`
+    );
+    const iceParameters = await this._protoo.request("restartIce", { transportId: transport.id });
+    await transport.restartIce({ iceParameters });
+  }
+
+  async recreateSendTransport(iceServers) {
+    this.emitRTCEvent("log", "RTC", () => `Recreating send transport ICE`);
+    await this.closeSendTransport();
+    await this.createSendTransport(iceServers);
+  }
 
   /**
    * Restart ICE in the underlying send peerconnection.
    */
   async restartSendICE(force = true) {
```

Comment:
Feels slightly weird to have this force argument. I think its in part a holdover from when this was being directly used as part of the watchdogs. Instead I would argue this function should always restart ice if called (or recreate the transport if its been closed I guess, still want to understand what that case is). Then we can handle any other logic about when we actually want to restart ice outside of this (which I think we are already doing in `checkXIceStatus`.

Reply:
Yep, that argument doesn't make much sense now, I'm going to remove it too.